src/resources/Fields/Fields.ts

```diff
@@ -0,0 +1,44 @@
+import API from '../../APICore';
+import Indexes from '../Indexes/Indexes';
+import Resource from '../Resource';
+import {FieldModel, ListFieldsParams} from './FieldsInterfaces';
+
+export default class Field extends Resource {
+    static baseUrl = `/rest/organizations/${API.orgPlaceholder}/indexes/fields`;
+
+    create(field: FieldModel) {
+        return this.api.post(Field.baseUrl, field);
+    }
+
+    createFields(fields: FieldModel[]) {
+        return this.api.post(`${Field.baseUrl}/batch/create`, fields);
+    }
+
+    deleteFields(fields: string[]) {
+        return this.api.delete(this.buildPath(`${Field.baseUrl}/batch/delete`, fields));
+    }
+
+    updateFields(fields: FieldModel[]) {
+        return this.api.put(`${Field.baseUrl}/batch/update`, fields);
+    }
+
+    synchronize() {
+        return this.api.post(`${Field.baseUrl}/synchronize`);
+    }
+
+    delete(fieldId: string) {
+        return this.api.delete(`${Field.baseUrl}/${fieldId}`);
+    }
+
+    get(fieldId: string) {
+        return this.api.get<FieldModel>(`${Field.baseUrl}/${fieldId}`);
+    }
+
+    update(fieldId: string, options: FieldModel) {
+        return this.api.put(`${Field.baseUrl}/${fieldId}`, options);
+    }
+
+    list(params: ListFieldsParams) {
+        return this.api.get<FieldModel[]>(this.buildPath(`${Indexes.baseUrl}/page/fields`, params));

```

Comment:
Its weird that this call doesn't return a `PageModel<FieldModel>`. Maybe we should enter them a bug, because we won't have a way to know how many pages there are :confused: 
```ts
interface PageModel<T = any> {
    items: T[];
    totalEntries: number;
    totalPages: number;
}
```

Reply:
😕  hmm I think it might be the problem in the swagger